indico/modules/events/management/forms.py

```diff
@@ -434,7 +434,23 @@ def __init__(self, source_event, target_event, set_defaults=False, **kwargs):
         super().__init__(**kwargs)
         self.selected_items.choices = [(option.name, option.friendly_name) for option in visible_options]
         # We disable all cloners that are not available, handle only cloning to new events,
-        # have conflicts with target_event or any of their dependencies has conflicts with target_event
-        self.selected_items.disabled_choices = [option.name for option in visible_options if not option.is_available
-                                                or conflicts[option.name]
-                                                or any(conflicts[dep] for dep in option.requires)]
+        # have conflicts with target_event or any of their dependencies have conflicts with target_event
+        disabled_choices = []
+        reasons = {}
+        for option in visible_options:
+            if not option.is_available:
+                disabled_choices.append(option.name)
+                reasons[option.name] = _('There is nothing to import')
+            elif conflicts[option.name]:
+                disabled_choices.append(option.name)
+                # Show only the first conflict
+                reasons[option.name] = conflicts[option.name][0]
+            else:
+                for dep in option.requires:
+                    if conflicts[dep]:
```

Comment:
I think `elif set(conflicts) & option.requires` does the same as these 3 lines and the break.

Please test it though, since I'm not sure what type `option.requires` is (might need an explicit `set()` as well if it's not a set)

Reply:
I'll check it :+1:

Reply:
And maybe this check should use `option.requires_deep`?

Reply:
Yes, I think we need `option.requires_deep` unless `visible_options` is topologically sorted (it isn't). We could have a situation where
`A -> B -> C` (`->` = depends on) and `C` has a conflict. If we process `A` before `B`, we won't disable `A`.

Reply:
Should be fixed now